core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessage.java

```diff
@@ -291,6 +295,9 @@ void notifySubscriber(SubscriptionImpl subscription, CompletableFuture<?> comple
             if (cause == null) {
                 try {
                     subscriber.onComplete();
+                } catch (Exception e) {
+                    logger.warn("Subscriber.onComplete() should not raise an exception. subscriber: {}",
```

Comment:
I think we should complete `completionFuture` with this error instead of treating this as a success scenario.